### PR TITLE
Adding LlamaIndex docs to integrations

### LINKED ISSUE #4798: [DOCS] Add integration docs for Argilla LlamaIndex
Add a section for the LlamaIndex Callback here https://docs.argilla.io/en/latest/tutorials_and_integrations/integrations/integrations.html

### PR DESCRIPTION
**Pull request by ignacioct**
Closes #4798

### CODE REVIEW

**Review by sdiazlor — COMMENTED** [review]
LGTM. Just a few comments to align it to the rest of the tutorials:

image

- [x] - If you could change the title
- [x] - Copy and paste the Running Argilla and Set up the Environment sections
- [x] -  Maybe an introduction is not needed, but a brief explanation of what LlamaIndex is or what it does.
- [x] - Could you remove the verbose outputs?
- [x] - Make a reference to how is the dataset we create.
- [x] - Review some typos
- [x] - Change the branch to main

**Comment by ignacioct** [comment]
Hi! Thanks for the comments @sdiazlor . I'll implement these comments tomorrow. One quick question tho, how would you name the notebook?

**Comment by sdiazlor** [comment]
> Hi! Thanks for the comments @sdiazlor . I'll implement these comments tomorrow. One quick question tho, how would you name the notebook?

I meant, for instance: `llamaindex`: Build and monitor LLMs applications

**Comment by ignacioct** [comment]
Gotcha! Everything should be ready now :) @sdiazlor

File: docs/_source/tutorials_and_integrations/integrations/llama_index.ipynb [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+   "outputs": [],
+   "source": [
+    "%pip install argilla-llama-index"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You will need to an Argilla Server running to monitor the LLM. You can either install the server locally or have it on HuggingFace Spaces. For a complete guide on how to install and initialize the server, you can refer to the [Quickstart Guide](https://docs.argilla.io/en/latest/getting_started/quickstart_installation.html)."
```
Comment (sdiazlor) [inline docs/_source/tutorials_and_integrations/integrations/llama_index.ipynb]: Could you just remove this that is repeated with the previous section?

**Review by sdiazlor — APPROVED** [review]

**Pull request merged**